crates/turbo-tasks/src/task_input.rs

```diff
@@ -860,3 +860,53 @@ impl TryFrom<&TaskInput> for RawVc {
         }
     }
 }
+
+macro_rules! tuple_impls {
+    ( $( $name:ident )+ ) => {
+        impl<$($name: Into<TaskInput>),+> From<($($name,)+)> for TaskInput {
+            #[allow(non_snake_case)]
+            fn from(s: ($($name,)+)) -> Self {
+                let ($($name,)+) = s;
+                let ($($name,)+) = ($($name.into(),)+);
+                TaskInput::List(vec![ $($name,)+ ])
+            }
+        }
+
+        impl<'a, $($name: FromTaskInput<'a, Error = anyhow::Error>,)+> FromTaskInput<'a> for ($($name,)+) {
+            type Error = anyhow::Error;
+
+            #[allow(non_snake_case)]
+            fn try_from(value: &'a TaskInput) -> Result<Self, Self::Error> {
+                match value {
+                    TaskInput::List(value) => {
+                        let mut i = 0;
+                        $(
+                            let $name = value.get(i).ok_or_else(|| anyhow!("missing tuple element"))?;
+                            let $name = FromTaskInput::try_from($name)?;
+                            #[allow(unused_assignments)]
+                            {
+                                i += 1;
+                            }
```

Comment:
You could use an iterator instead, that would avoid the unused `i += 1`.

let mut iter = value.iter();
$(
  iter.next() ...
  ...
```